packages/invoices/src/OutboundShipment/DetailView/modals/ItemDetailsModal.tsx

```diff
@@ -86,18 +100,83 @@ const useBatchRows = (summaryItem: OutboundShipmentSummaryItem | null) => {
     if (!data) return;
 
     setBatchRows(() => {
-      return data
+      const rows = data
         .map(batch => {
           const matchingInvoiceRow = Object.values(summaryItem.batches).find(
             ({ stockLineId }) => stockLineId === batch.id
           );
           return { ...batch, quantity: matchingInvoiceRow?.numberOfPacks ?? 0 };
         })
         .sort(sortByDisabledThenExpiryDate);
+      rows.push(createPlaceholderRow());
+
+      return rows;
     });
   }, [data]);
 
-  return { batchRows, isLoading };
+  return { batchRows, isLoading, setBatchRows };
+};
+
+export type PackSizeController = ReturnType<typeof usePackSizeController>;
+
+const usePackSizeController = (batches: { packSize: number }[]) => {
+  // Creating a sorted array of distinct pack sizes
+  const packSizes = Array.from(
+    new Set(
+      batches
```

Comment:
perhaps it would be good to filter batches that have no stock or are on hold. 
we aren't saving the requested pack size, so I'm wondering if there's value in multiplying quantity*pack size and popping that in the placeholder row 🤷

Reply:
Ah yes good point. I suppose that's why you can add multiple placeholders in mSupply with different pack sizes? For all the effort we're putting into FEFOing with the right pack size I wonder why we aren't doing similar with supporting placeholders with the right pack size 🤔